java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -881,6 +899,45 @@ private boolean isCheckedXPFlagName(ExpressionTree tree) {
   public Description matchMethod(MethodTree tree, VisitorState state) {
     if (disabled) return Description.NO_MATCH;
 
+    if (!configTestMethodProperties.isEmpty() && tree != null && tree.getBody() != null && tree.getBody().getStatements() != null) {
+      List<? extends StatementTree> bt = tree.getBody().getStatements();
+      for (StatementTree st : bt) {
+        if (st != null && st.getKind().equals(Kind.EXPRESSION_STATEMENT)) {
+          ExpressionStatementTree est = (ExpressionStatementTree) st;
```

Comment:
1. Shouldn't the statement count here be just one? Can we avoid the loop? 
2. How about calling `evalExpr(st)`? If not, `evalExpr(mit)` below and use the result of that. 
3. Before calling the`evalExpr`, call `getXPAPI` and see whether the API is `Mock` method ? You may also want to update `getXPAPI` to do the necessary processing to check for `when` and return the appropriate type. 

So, the code will be: 
```
API api = getXPAPI(mit) // may be st too.
if(api.equals(API.MOCK)) {
     Value value = evalExpr(mit) // may be st too and update evalExpr
      if(value != Value.BOT) {
         // code starting at line 922. 
      }
}
```